src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/updateotherparty/UpdateOtherPartyAboutToSubmitHandlerTest.java

```diff
@@ -107,52 +113,161 @@ public void givenOtherPartiesUcbIsYes_thenUpdateCaseDataOtherPartyUcb() {
 
     @Test
     public void givenNewOtherPartyAdded_thenAssignAnId() {
-        sscsCaseData.setOtherParties(Arrays.asList(buildOtherPartyWithAppointeeAndRep(null, null, null)));
+        sscsCaseData.setOtherParties(Arrays.asList(
+            buildOtherPartyWithAppointeeAndRep(null, null, null)));
 
         PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
-        assertEquals(1, response.getData().getOtherParties().size());
-        assertEquals("1", response.getData().getOtherParties().get(0).getValue().getId());
-        assertEquals("2", response.getData().getOtherParties().get(0).getValue().getAppointee().getId());
-        assertEquals("3", response.getData().getOtherParties().get(0).getValue().getRep().getId());
-        assertTrue(YesNo.isYes(response.getData().getOtherParties().get(0).getValue().getSendNewOtherPartyNotification()));
+
+        List<CcdValue<OtherParty>> otherParties = response.getData().getOtherParties();
+
+        Assertions.assertThat(otherParties)
+            .hasSize(1)
+            .extracting(CcdValue::getValue)
+            .extracting(OtherParty::getSendNewOtherPartyNotification)
+            .containsOnly(YES);
+
+        Assertions.assertThat(otherParties)
+            .extracting(CcdValue::getValue)
+            .flatExtracting(otherParty -> List.of(otherParty.getAppointee(), otherParty.getRep()))
+            .extracting(Entity::getId)
+            .hasSize(2)
+            .doesNotContainNull()
+            .extracting(String::length)
+            .containsOnly(36);
+
         assertEquals(0, response.getErrors().size());
     }
 
     @Test
     public void givenExistingOtherParties_thenNewOtherPartyAssignedNextId() {
-        sscsCaseData.setOtherParties(Arrays.asList(buildOtherParty("2"), buildOtherParty("1"), buildOtherPartyWithAppointeeAndRep(null, null, null)));
+        sscsCaseData.setOtherParties(Arrays.asList(
```

Comment:
I don't think we need tests which regard assigning next ids since we're switching to UUIDs

Reply:
Renamed the tests